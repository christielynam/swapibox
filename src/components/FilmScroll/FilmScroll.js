import React from 'react';
import styled from 'styled-components';

const FilmContainer = styled.div`
  width: 90%;
  height: 700px;
  margin: auto;
  background-color: white;
  h3 {
    padding: 20px;
    font-size: 35px;
  };
  p {
    padding: 0 20px 0 20px;
    font-size: 20px;
  };
  h6 {
    font-size: 25px;
  }
`

const FilmScroll = ({film}) => {
  return (
    <FilmContainer>
      <h3>{film.title}</h3>
      <p>{film.opening_crawl}</p>
      <h6>{film.release_date}</h6>
    </FilmContainer>
  )
}

export default FilmScroll