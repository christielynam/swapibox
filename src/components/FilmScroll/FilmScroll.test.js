import React from 'react';
import FilmScroll from './FilmScroll';
import { shallow } from 'enzyme'

describe('FilmScroll', () => {
  let wrapper
  let mockFilm

  beforeEach(() => {
    mockFilm = {
      title: 'Some StarWars Title',
      opening_crawl: 'Some details about the movie',
      release_date: 'Movie release date'
    }
    wrapper = shallow(<FilmScroll film={mockFilm} />)
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})