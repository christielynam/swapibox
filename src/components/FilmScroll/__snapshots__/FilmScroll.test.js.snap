// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmScroll matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilmScroll
    film={
      Object {
        "opening_crawl": "Some details about the movie",
        "release_date": "Movie release date",
        "title": "Some StarWars Title",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h3>
          Some StarWars Title
        </h3>,
        <p>
          Some details about the movie
        </p>,
        <h6>
          Movie release date
        </h6>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Some StarWars Title",
        },
        "ref": null,
        "rendered": "Some StarWars Title",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Some details about the movie",
        },
        "ref": null,
        "rendered": "Some details about the movie",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Movie release date",
        },
        "ref": null,
        "rendered": "Movie release date",
        "type": "h6",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  width: 90%;
  height: 650px;
  margin: auto;
  background-color: white;
  h3 {
    padding: 20px;
    font-size: 35px;
  };
  p {
    padding: 0 20px 0 20px;
    font-size: 20px;
  };
  h6 {
    font-size: 25px;
  }
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h3>
            Some StarWars Title
          </h3>,
          <p>
            Some details about the movie
          </p>,
          <h6>
            Movie release date
          </h6>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Some StarWars Title",
          },
          "ref": null,
          "rendered": "Some StarWars Title",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Some details about the movie",
          },
          "ref": null,
          "rendered": "Some details about the movie",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Movie release date",
          },
          "ref": null,
          "rendered": "Movie release date",
          "type": "h6",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  width: 90%;
  height: 650px;
  margin: auto;
  background-color: white;
  h3 {
    padding: 20px;
    font-size: 35px;
  };
  p {
    padding: 0 20px 0 20px;
    font-size: 20px;
  };
  h6 {
    font-size: 25px;
  }
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
